analyzer/src/main/kotlin/curation/ClearlyDefinedPackageCurationProvider.kt

```diff
@@ -96,6 +96,7 @@ class ClearlyDefinedPackageCurationProvider(
         val (type, provider) = pkgId.toClearlyDefinedTypeAndProvider() ?: return emptyList()
         val namespace = pkgId.namespace.takeUnless { it.isEmpty() } ?: "-"
 
+        @Suppress("TooGenericExceptionCaught")
```

Comment:
Hint: You could avoid this suppression by using `runCatching` / `onFailure` / `when` like

        val curation = runCatching {
            // TODO: Maybe make PackageCurationProvider.getCurationsFor() a suspend function; then all derived
            //       classes could deal with coroutines more easily.
            runBlocking(Dispatchers.IO) { service.getCuration(type, provider, namespace, pkgId.name, pkgId.version) }
        }.onFailure { e ->
            when (e) {
                is HttpException -> {
                    // A "HTTP_NOT_FOUND" is expected for non-existing curations, so only handle other codes as a failure.
                    if (e.code() != HttpURLConnection.HTTP_NOT_FOUND) {
                        e.showStackTrace()

                        log.warn {
                            val message = e.response()?.errorBody()?.string() ?: e.collectMessagesAsString()
                            "Getting curations for '${pkgId.toCoordinates()}' failed with code ${e.code()}: $message"
                        }
                    }
                }

                is JsonMappingException -> {
                    e.showStackTrace()

                    log.warn { "Deserializing the ClearlyDefined curation for '${pkgId.toCoordinates()}' failed." }
                }

                else -> {
                    e.showStackTrace()

                    log.warn {
                        "Querying ClearlyDefined curation for '${pkgId.toCoordinates()}' failed: ${e.collectMessagesAsString()}"
                    }
                }
            }
        }.getOrNull() ?: return emptyList()

Reply:
Yes, this looks good. Changed to this construct.